### ISSUE TITLE
Using validation rules in a sub-array makes the field required

### ISSUE THREAD

**Issue by jormaechea** [vlucas/valitron#271]
**Use case**
Validate an optional field within a numeric array of associative arrays.

**Example**

Normally, the `dateFormat` rule allows `null` value (which is correct, because it's not required).
This is valid: 
```php
$validator = new Valitron\Validator(["dateCreated" => "2018-12-11T13:52:20-03:00"]);
$validator->rule("dateFormat", "dateCreated", "Y-m-d\TH:i:sP");
```

But when the date field is within an array, it automatically makes it required.
This is invalid:
```php
$validator = new Valitron\Validator([
  [ "dateCreated" => null ]
]);
$validator->rule("optional", "*.dateCreated"); // This line doesn't affect Valitron's behaviour
$validator->rule("dateFormat", "*.dateCreated", "Y-m-d\TH:i:sP");
```

**Expected behaviour**
Valitron should check the `dateFormat` rule in the same way whether or not it's in an array, which means, to treat the `dateCreated` field as optional.

**Notes**
This also happens, for example with `numeric`, `integer`, `lengthMin` and other rules.

**Comment by willemwollebrants** [comment]
I think (and testing seems to confirm it) this is already fixed in the dev-master branch. Can you please try it with ```composer require vlucas/valitron:dev-master```

If it's fixed there I'll make a new release. If it's not, I'll try and fix it first :)

**Comment by jormaechea** [comment]
Yes, it's working on `dev-master`!! I'll wait till the new release. :smile:
